### ISSUE TITLE
[semi] not detected in interface/object type definitions

### ISSUE THREAD

**Issue by Bkucera** [typescript-eslint/typescript-eslint#610]
Semicolons are not detected in interface/object type definitions:

```JSON
{
  "rules": {
	"semi": "off",
    "@typescript-eslint/semi": ["error", "never"]
  }
}
```

```TS
type foo = {
  bar:string;
  baz:number;
}; // <--- only this semicolon is caught
```

**Expected Result**
should catch semicolon on lines 2 & 3
**Actual Result**
only semi on line 4 is caught

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.10.3-alpha.1` |
| `@typescript-eslint/parser`        | `1.10.3-alpha.1` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `8.9.3` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
Semicolons are not caught inside the type definition because they are not statement.
Additionally the syntax for type member delimiters is different - it allows semis, commas or nothing.

If you're looking to lint them, consider using [member-delimiter-style](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-delimiter-style.md)

**Comment by Bkucera** [comment]
@bradzacher oh awesome thanks! I was confused because `tslint` does lint those `semi` with the `semicolon` rule.
